model_compression_toolkit/core/keras/mixed_precision/configurable_weights_quantizer.py

```diff
@@ -0,0 +1,154 @@
+                    use this quantizer.
+                float_weights: Float weights of the layer.
+                max_candidate_idx: Index of the node's candidate that has the maximal bitwidth (must exist absolute max).
+
+            """
+
+            super(ConfigurableWeightsQuantizer, self).__init__()
+
+            self.node_q_cfg = node_q_cfg
+            self.float_weights = float_weights
+            self.max_candidate_idx = max_candidate_idx
+
+            # Make sure the candidates configurations arrived in descending order.
+            curmax = (np.inf, np.inf)
+            n_candidate_bits = [(x.weights_quantization_cfg.weights_n_bits, x.activation_quantization_cfg.activation_n_bits)
+                                for x in self.node_q_cfg]
+            for candidate_bits in n_candidate_bits:
+                assert candidate_bits < curmax
+                curmax = candidate_bits
+
+            for qc in self.node_q_cfg:
+                if qc.weights_quantization_cfg.enable_weights_quantization != \
+                        self.node_q_cfg[0].weights_quantization_cfg.enable_weights_quantization:
+                    Logger.error("Candidates with different weights enabled properties is currently not supported.")
+
+            # Initialize quantized weights for each weight that should be quantized.
+            self.quantized_weights = []
+            for qc in self.node_q_cfg:
+                qc_weights = qc.weights_quantization_cfg
+                q_weight = qc_weights.weights_quantization_fn(self.float_weights,
+                                                              qc_weights.weights_n_bits,
+                                                              True,
+                                                              qc_weights.weights_quantization_params,
+                                                              qc_weights.weights_per_channel_threshold,
+                                                              qc_weights.weights_channels_axis)
+
+                self.quantized_weights.append(tf.convert_to_tensor(q_weight, dtype=tf.float32))
+
+            self.active_quantization_config_index = self.max_candidate_idx
+
+        def set_weights_bit_width_index(self,
+                                        index: int):
+            """
+            Change the "active" bitwidth index the configurable quantizer uses, so a different quantized weight
+            will be used.
+
+            Args:
+                index: Quantization configuration candidate index to use.
+
+            """
+
+            assert index < len(self.node_q_cfg), \
+                f'Quantizer has {len(self.node_q_cfg)} ' \
+                f'possible nbits. Can not set index {index}'
+            self.active_quantization_config_index = index
+
+        def __call__(self,
+                     inputs: tf.Tensor,
+                     training: bool) -> tf.Tensor:
```

Comment:
why do you need the training flag for?

Reply:
We don't, it's a mistake from copying the code from the previous version of this class.